### DISCUSSION TITLE
How to send synchronous commands to multiple crazyflies

### DISCUSSION THREAD

**Discussion by Jushan-Randy-Chen** [USC-ACTLab/crazyswarm#678] [Q&A] [answered]
I wonder how I can send commands to multiple crazyflies synchronously? Right now I am trying to control 3 drones, and I am sending position commands to each of them sequentially :
```
`
swarm = Crazyswarm()
timeHelper = swarm.timeHelper
cf1 = swarm.allcfs.crazyflies[0]
cf2 = swarm.allcfs.crazyflies[1]
cf3 = swarm.allcfs.crazyflies[2]

cf1.goTo(cf1_takeoff_pos, yaw=0.0, duration=3.0)

cf2.goTo(cf2_takeoff_pos, yaw=0.0, duration=3.0)

cf3.goTo(cf3_takeoff_pos, yaw=0.0, duration=3.0)

`
```
How can I make these 3 crazyflies go to their respective take-off positions at the same time, rather than executing the commands sequentially?

**Comment by whoenig** [comment, marked as answer]
This will actually execute it almost at the same time, since the goTo command does not block.

If you need exact synchronization, you have to use broadcast commands (where each drone receives the exact same command). For goTo, the broadcast command can only support relative and not absolute positions, so it doesn't work in your case.
